docroot/sites/all/conf/environment.inc

```diff
@@ -36,8 +36,18 @@ elseif (getenv('DOCKER_LOCAL')) {
   $platform = D_PLATFORM_DOCKER;
 }
 
-if (!empty($env) && !empty($platform) && !empty($instance)) {
+if (!empty($env) && !empty($platform)) {
+  if (empty($instance) && php_sapi_name() === 'cli') {
+    if (function_exists('drupal_set_message')) {
+      drupal_set_message('SETTINGS_INSTANCE is not set and we were not invoked from the CLI.', 'warning');
+    }
+  }
+  define('SETTINGS_PLATFORM', $platform);
   define('SETTINGS_ENVIRONMENT', $env);
   define('SETTINGS_INSTANCE', $instance);
```

Comment:
Do we not still need to check that $instance is not empty before we do this?

Reply:
I don't see why.
What's the benefit?